### PR TITLE
[Snappi] Fix potential infinite loop in get_ipv6_addrs_in_subnet

### PR DESCRIPTION
**Pull request by sdszhang**
### Description of PR
Summary:
Address review comments from #23037 (by @StormLiangMS):
Rewrite `get_ipv6_addrs_in_subnet` to use sequential host iteration (for loop) instead of random address generation, consistent with `get_addrs_in_subnet` behavior. This eliminates the potential infinite loop on small subnets and simplifies the implementation.

### Type of change
- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] New Test case
    - [ ] Skipped for non-supported platforms
- [ ] Test case improvement

### Back port request
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405
- [ ] 202411
- [ ] 202505
- [ ] 202511

#### What is the motivation for this PR?
Address review feedback from PR #23037: the `get_ipv6_addrs_in_subnet` function used random address generation in a while loop, which could infinite-loop if the requested number of IPs exceeds what the subnet can provide after exclusions.

#### How did you do it?
Replaced the random generation while loop with sequential `network.hosts()` iteration and a for loop, matching the pattern used by `get_addrs_in_subnet` for IPv4. This naturally terminates when the subnet runs out of hosts.

#### How did you verify/test it?
Code review and local validation.

#### Any platform specific information?
N/A

#### Supported testbed topology if it is a new test case?
N/A

### Documentation
N/A

### CODE REVIEW

File: tests/common/snappi_tests/common_helpers.py [hunk 1]
```diff
@@ -905,15 +905,21 @@ def get_ipv6_addrs_in_subnet(subnet, number_of_ip, exclude_ips=None):
...
     subnet = subnet.encode().decode("utf-8")
     ipv6_list = []
+    seen = set()
     exclude_set = set(exclude_ips) if exclude_ips else set()
-    while len(ipv6_list) < number_of_ip:
+    # Avoid infinite loop when IPv6 subnet is smaller than requested IPs
+    max_attempts = number_of_ip * 10
+    attempts = 0
+    while len(ipv6_list) < number_of_ip and attempts < max_attempts:
+        attempts += 1
```
[the remark this answers is not in the document]
Comment (sdszhang) [reply tests/common/snappi_tests/common_helpers.py]: added checks upfront.

File: tests/common/snappi_tests/common_helpers.py [hunk 2]
```diff
@@ -904,16 +904,28 @@ def get_ipv6_addrs_in_subnet(subnet, number_of_ip, exclude_ips=None):
...
+    pytest_assert(available >= number_of_ip,
+                  "Not enough addresses in subnet {} to satisfy request for {} IPs "
+                  "({} available after exclusions)".format(subnet, number_of_ip, available))
+
+    ipv6_list = []
+    seen = set()
+    # Avoid infinite loops in corner cases
+    max_attempts = number_of_ip * 10
+    attempts = 0
+    while len(ipv6_list) < number_of_ip and attempts < max_attempts:
```
Comment (ediwibowo-msft) [inline tests/common/snappi_tests/common_helpers.py]: The capped retry ensures the `while` loop terminates, but it does not guarantee success once `max_attempts` is reached.

Should the `while` loop  be converted to a `for` loop to avoid a potential infinite loop?

For example:

```
    results = []

    for addr in net.hosts():
        addr_str = str(addr)
        if addr_str in exclude_set:
            continue

        results.append(addr_str)
        if len(results) == number_of_ip:
            break
    else:
        raise ValueError(
            "Not enough addresses in subnet {} to satisfy request for {} IPs".format(
                subnet, number_of_ip
            )
        )

    return results
```

Comment (sdszhang) [reply tests/common/snappi_tests/common_helpers.py]: Thanks for the comment, that's the other option to solve this issue.  For now, would prefer to keep the behavior unchanged, just fix the potential loop issue.

Comment (ediwibowo-msft) [reply tests/common/snappi_tests/common_helpers.py]: A potential issue is that the loop can stop because `attempts < max_attempts` becomes false before `len(ipv6_list) == number_of_ip`

Comment (sdszhang) [reply tests/common/snappi_tests/common_helpers.py]: updated

File: tests/common/snappi_tests/common_helpers.py [hunk 3]
```diff
@@ -904,18 +904,15 @@ def get_ipv6_addrs_in_subnet(subnet, number_of_ip, exclude_ips=None):
...
-            continue
-        ipv6_list.append(addr_str)
 
+    ipv6_list = []
+    for addr in network.hosts():
+        if str(addr) not in exclude_set:
+            ipv6_list.append(str(addr))
+            if len(ipv6_list) == number_of_ip:
+                break
     return ipv6_list
```
Comment (ediwibowo-msft) [inline tests/common/snappi_tests/common_helpers.py:916]: Should an exception be raised if `len(ipv6_list) < number_of_ip`?

Comment (sdszhang) [reply tests/common/snappi_tests/common_helpers.py:916]: IPv4 has the same behavior. we can keep it this way.
